test/functional/wallet_balance.py

```diff
@@ -296,5 +303,27 @@ def test_balances(*, fee_node_1=0):
             assert_equal(self.nodes[0].getbalances()['mine']['untrusted_pending'], Decimal('0.1'))
 
 
+        # Tests the lastprocessedblock JSON object in getbalances, getwalletinfo
+        # and gettransaction by checking for valid hex strings and by comparing
+        # the hashes & heights between generated blocks.
+        self.log.info("Test getbalances returns expected lastprocessedblock json object")
+        prev_hash = self.nodes[0].getbestblockhash()
+        prev_height = self.nodes[0].getblock(prev_hash)['height']
+        self.generatetoaddress(self.nodes[0], 5, self.nodes[0].get_deterministic_priv_key().address)
+        lastblock = self.nodes[0].getbalances()['lastprocessedblock']
+        assert_is_hash_string(lastblock['hash'])
+        assert_equal((prev_hash == lastblock['hash']), False)
+        assert_greater_than(lastblock['height'], prev_height)
+
+        self.log.info("Test getwalletinfo returns expected lastprocessedblock json object")
+        walletinfo = self.nodes[0].getwalletinfo()
+        assert_greater_than(walletinfo['lastprocessedblock']['height'], 0)
+        assert_is_hash_string(walletinfo['lastprocessedblock']['hash'])
+
+        self.log.info("Test gettransaction returns expected lastprocessedblock json object")
```

Comment:
should also have a test on the height to ensure that this is in fact the "expected" `lastprocessedblock` json object?

Reply:
Done.